File: 移动元素.py
"""
双指针 一开始一直想着如何删除多余部分，然后发现有一句无需考虑超出长度
"""
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        lastNoneValPoint = 0
        for i in range(len(nums)):
            if nums[i] != val:
                nums[lastNoneValPoint] = nums[i]
                lastNoneValPoint += 1
        return lastNoneValPoint
                
            

File: 移动元素.cpp
/*
* 双指针
*/
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int lastNoneValPoint = 0;
        for(int i = 0; i < nums.size(); i++)
        {
            if(nums[i] != val)
            {
                nums[lastNoneValPoint++] = nums[i];
            }
        }
        return lastNoneValPoint;
    }
};
